### PR TITLE
initial contribution docs

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary

initial contribution docs

### CODE REVIEW

File: website/docs/getting-started/04--contribution.md
Comment (heatherlarsen) [inline website/docs/getting-started/04--contribution.md:15]: This is just for consideration, but putting what we don't accept first gets to the meat of the "problem" faster so consumers don't have to search for it.

```suggestion
While we don't accept direct contributions of new components or patterns, we do welcome contributions in the following forms:

* [Reporting bugs](https://docs.google.com/forms/d/e/1FAIpQLScpMXgrUTVT5fYriu4Pp48r4Nl_eCPluVnJLg0Yg3NXsRWvIA/viewform) and [giving feedback](https://docs.google.com/forms/d/e/1FAIpQLSdPShbSiUYgimppZeqWT99bVi6zLbKDThQUQTirH3RCj2okdg/viewform)
* [Proposing new components or patterns](https://docs.google.com/forms/d/e/1FAIpQLScpMXgrUTVT5fYriu4Pp48r4Nl_eCPluVnJLg0Yg3NXsRWvIA/viewform)
* Improving our documentation
* Participating in discussions or coming to Office Hours

We continue to listen to feedback as we consider the best way to open contributions further.
```

File: website/docs/getting-started/04--contribution.md
Comment (heatherlarsen) [inline website/docs/getting-started/04--contribution.md:3]: I don't think this page needs a description; it doesn't feel like it's providing additional value.

```suggestion
```

**Review by heatherlarsen — CHANGES_REQUESTED** [review]
I added a couple of suggestions, but also think this page is missing the Flight icon [contribution details](https://flight-hashicorp.vercel.app/design#contributing).

**Comment by Dhaulagiri** [comment]
@heatherlarsen those will exist soon under the icons page. We could link out to those from here when they exist or consider moving them here. What do you think?

**Comment by heatherlarsen** [comment]
@Dhaulagiri They should exist here and be linked to from the Icon pages. All contribution documentation should live in one place for easier access. 

cc @cveigt

**Comment by Dhaulagiri** [comment]
I'll pull them out of https://github.com/hashicorp/design-system/pull/735 and move them here and let y'all know when it can be reviewed again

File: website/docs/getting-started/contribution/partials/icons.md
Comment (Dhaulagiri) [inline website/docs/getting-started/contribution/partials/icons.md:33]: @didoo I see these images in folders named after their path in markdown. does it make sense for me to create a `/assets/getting-started/contribution/icons` folder to house these images and move them over?

(I know i have a homework assignment for this but I haven't gotten to it yet)

**Comment by Dhaulagiri** [comment]
@hashicorp/hds-design take a look now and let me know what you think. i'm not totally sure how best to structure the content here but we can fix that once we have better guidelines in place (i think).

Note: some images aren't appearing on the icon section but this existed since this was brought over. I'll open a separate bug to fix that.

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [inline website/docs/getting-started/contribution/index.md:10]: I'm not sure if I did the anchor link correctly.

```suggestion
* [Proposing new components or patterns](https://docs.google.com/forms/d/e/1FAIpQLScpMXgrUTVT5fYriu4Pp48r4Nl_eCPluVnJLg0Yg3NXsRWvIA/viewform)
* [Add an icon](#iconcontributions)
```

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [inline website/docs/getting-started/contribution/index.md:16]:
```suggestion
## Code contributions 
```

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [inline website/docs/getting-started/contribution/index.md:18]:
```suggestion
For contributions to the individual code packages, consult the `CONTRIBUTING.md` files.
```

**Review by heatherlarsen — COMMENTED** [review]
Thanks for getting that added! 

I included a few more minor suggestions. I think we can do a bit more work to clean up the Icon contribution section, but I'm unsure how to add comments for those.

File: website/docs/getting-started/contribution/index.md
Comment (didoo) [inline website/docs/getting-started/contribution/index.md:18]: I would change the order (tokens, icons, components)

Also, I would add a follow-up ticket post launch, to review this content (as is now, is pretty useless for a potential "contributor" and seems to contradict what stated above that we don't accept contributions on components

**Review by didoo — APPROVED** [review]

File: website/docs/getting-started/contribution/index.md
Comment (Dhaulagiri) [reply website/docs/getting-started/contribution/index.md:18]: > seems to contradict what stated above that we don't accept contributions on components

we dont't accept new components, but those files include instructions for how to work on the projects if they wanted to contribute bug fixes. in either case, we do need to flesh this out more in time.

File: website/docs/getting-started/contribution/index.md
Comment (Dhaulagiri) [reply website/docs/getting-started/contribution/index.md:10]: I accepted this without looking closely enough. I don't think this will work but I'll try to figure it out.

Comment (Dhaulagiri) [reply website/docs/getting-started/contribution/index.md:10]: `#icon-contributions` was what worked

File: website/docs/getting-started/contribution/index.md
Comment (Dhaulagiri) [reply website/docs/getting-started/contribution/index.md]: updated the order

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [reply website/docs/getting-started/contribution/index.md:10]: Perfect! Thanks for confirming; I was seeing conflicting information on the google.

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [reply website/docs/getting-started/contribution/index.md]: Yea, our support workflow can use a bit of work. The one I linked is currently being used for new requests, change requests, and bugs (at least those found in Figma), but I think the update you made works until we have this more sorted out.

**Comment by Dhaulagiri** [comment]
@heatherlarsen since you requested changes I need a re-approval before I can merge

File: website/docs/getting-started/contribution/index.md
Comment (heatherlarsen) [reply website/docs/getting-started/contribution/index.md]: @didoo @Dhaulagiri What about adding some language before "For contributions to the individual code packages" around , something like: 

"We also accept minor contributions to existing code packages, such as fixing a bug, proposing a code refactor, etc. For contributions to the individual code packages..."

**Review by heatherlarsen — APPROVED** [review]
Looks good for now. 👍🏻

**Pull request merged**
